src/Juvix/Core/IR/TransformExt.hs

```diff
@@ -0,0 +1,115 @@
+  -> (XBound  ext1 primTy primVal -> XBound  ext2 primTy primVal)
+  -> (XFree   ext1 primTy primVal -> XFree   ext2 primTy primVal)
+  -> (XPrim   ext1 primTy primVal -> XPrim   ext2 primTy primVal)
+  -> (XApp    ext1 primTy primVal -> XApp    ext2 primTy primVal)
+  -> (XAnn    ext1 primTy primVal -> XAnn    ext2 primTy primVal)
+  -> (TermX   ext1 primTy primVal -> TermX   ext2 primTy primVal)
+  -> (ElimX   ext1 primTy primVal -> ElimX   ext2 primTy primVal)
+  -> ExtTransformTE ext1 ext2 primTy primVal
+pattern ExtTransformTE { etStar, etPrimTy, etPi, etLam, etElim, etBound,
+                         etFree, etPrim, etApp, etAnn, etTermX, etElimX } =
+  ExtTransformTEF {
+    etfStar   = Coerce etStar,
+    etfPrimTy = Coerce etPrimTy,
+    etfPi     = Coerce etPi,
+    etfLam    = Coerce etLam,
+    etfElim   = Coerce etElim,
+    etfBound  = Coerce etBound,
+    etfFree   = Coerce etFree,
+    etfPrim   = Coerce etPrim,
+    etfApp    = Coerce etApp,
+    etfAnn    = Coerce etAnn,
+    etfTermX  = Coerce etTermX,
+    etfElimX  = Coerce etElimX
+  }
+
+
+extTransformTF :: Applicative f
+               => ExtTransformTEF f ext1 ext2 primTy primVal
+               -> Term' ext1 primTy primVal -> f (Term' ext2 primTy primVal)
+extTransformTF fs (Star' i e)   = Star' i <$> etfStar fs e
+extTransformTF fs (PrimTy' k e) = PrimTy' k <$> etfPrimTy fs e
+extTransformTF fs (Pi' π s t e) = Pi' π <$> extTransformTF fs s
+                                        <*> extTransformTF fs t
+                                        <*> etfPi fs e
+extTransformTF fs (Lam' t e)    = Lam' <$> extTransformTF fs t <*> etfLam fs e
+extTransformTF fs (Elim' f e)   = Elim' <$> extTransformEF fs f <*> etfElim fs e
+extTransformTF fs (TermX e)    = TermX <$> etfTermX fs e
+
+extTransformT :: ExtTransformTE ext1 ext2 primTy primVal
+              -> Term' ext1 primTy primVal -> Term' ext2 primTy primVal
+extTransformT fs t = runIdentity $ extTransformTF fs t
+
+extTransformEF :: Applicative f
+               => ExtTransformTEF f ext1 ext2 primTy primVal
+               -> Elim' ext1 primTy primVal -> f (Elim' ext2 primTy primVal)
+extTransformEF fs (Bound' x e) = Bound' x <$> etfBound fs e
+extTransformEF fs (Free' x e) = Free' x <$> etfFree fs e
+extTransformEF fs (Prim' k e) = Prim' k <$> etfPrim fs e
+extTransformEF fs (App' f s e) = App' <$> extTransformEF fs f
+                                      <*> extTransformTF fs s
+                                      <*> etfApp fs e
+extTransformEF fs (Ann' π s t e) = Ann' π <$> extTransformTF fs s
+                                          <*> extTransformTF fs t
+                                          <*> etfAnn fs e
+extTransformEF fs (ElimX e) = ElimX <$> etfElimX fs e
+
+extTransformE :: ExtTransformTE ext1 ext2 primTy primVal
+              -> Elim' ext1 primTy primVal -> Elim' ext2 primTy primVal
+extTransformE fs t = runIdentity $ extTransformEF fs t
+
```

Comment:
These all seem to fall out from the ADT itself.

Could we somehow do a deriving via (not sure if it can get the fields this way!). To basically auto generate this.

Would be useful if we use this technique in more places.